src/librustc_hir/definitions.rs

```diff
@@ -95,11 +91,6 @@ pub struct Definitions {
     /// Item with a given `LocalDefId` was defined during macro expansion with ID `ExpnId`.
     expansions_that_defined: FxHashMap<LocalDefId, ExpnId>,
     next_disambiguator: FxHashMap<(LocalDefId, DefPathData), u32>,
```

Comment:
Could you explain in a bit more details what you have in mind? `next_disambiguator` is used in `create_def` to find the next free disambiguator.

Reply:
It can be moved out from `Defintions::create_def` to `Resolver::create_def_with_parent` in the same way as other code that was moved in that direction in this PR.

Reply:
As a side note, `Resolver::create_def_with_parent` can also be renamed to `Resolver::create_def`.

Reply:
So if I understand correctly, `Definitions::create_def` would take a `DefKey` as argument and that `key` would be computed beforehand in `Resolver::create_def`?

Reply:
`DefKey`, or just a `disambiguator` which should be enough.

Reply:
I have mixed feeling about this. On one side, its removes one more field that is not used after lowering but on the other side, it makes `Definitions::create_def` API more complicated and more error prone. Technically, one could create a `disambiguator` for a given parent/data couple but call (by mistake) `Definitions::create_def` with another unrelated parent/data couple.

Reply:
It could take a reference to `next_disambiguator` then, or a closure that returns the disambiguator given `parent` and `data`.

Reply:
Thank you, I like the closure approach very much! I've just pushed another commit that moves `next_disambiguator` to the `Resolver`